### DISCUSSION TITLE
can't pass cmake with no errors on windows 10 please help !

### DISCUSSION THREAD

**Discussion by ytf11** [vsg-dev/VulkanSceneGraph#717] [Q&A]
tried different versions of cmake and vulkan though

**Comment by sbrkopac** [comment]
> tried different versions of cmake and vulkan though

VSG compiles out of the box with cmake. You can look at the CI for examples. You'll have to post more information if you want help.

**Reply by ytf11** [reply]
Thank @sbrkopac for ur response ! You let me know I am not careful enough lol.  I am a new guy for VSG but I am serious and want to use it !!! 

Thanks again

**Comment by robertosfield** [comment]
@ytf11 you don't provide any information about what is going wrong so it's impossible for us to give you any specific guidance.

You'll need to provide us with more information about what steps you trying and what results you are getting to help us diagnose what you are doing wrong and what to do different.

The guide to build on Windows can be found in the INSTALL.md's Window section so I recommend you follow this and let us know what point things go wrong for you:

https://github.com/vsg-dev/VulkanSceneGraph/blob/master/INSTALL.md#detailed-instructions-for-setting-up-your-environment-and-building-for-microsoft-windows

**Reply by ytf11** [reply]
@robertosfield thanks a lot for so quick responses!

following ur instructions,
I actually have cloned vsg-dev/VulkanSceneGraph and run cmake successfully on windows 10 !!!
From the created .sln, i could build 2 projects, ALL-BUILD and INSTALL well. They looked successful and
vsg stuff was copied to c:\program files\vsg (2 sub-directories inside, include/ and lib/).
In a word, I believe vsg works already! CMAKE_PREFIX_PATH = C:\Program Files\VSG was set in environment variables!

Thanks again

**Reply by ytf11** [reply]
@robertosfield, 

I am also trying vsg-dev/vsgExamples.git and here's what its about, 

the following works,

git clone https://github.com/vsg-dev/vsgExamples.git
cd vsgExamples
cmake .

but i could NOT run make -j 8 'cause of missing makefile! my cmd display looks like,

/c/VSG/vsgExamples>make -j 8
make: *** No targets specified and no makefile found. Stop.
/c/VSG/vsgExamples>

I could open cmake-created .sln, built "ALL-BUILD" then "INSTALL". Now i can run the created excutables well in cmd window, for example, vsgdraw.exe runs Okay !

One problem is i can NOT run them, vsgdraw for example, in Debugger(VS). i tracked down to main -> vsg::getEnvPaths() -> getenv_s() and got trouble !!! getenv_s() can't get environment variable correctly though I have set VSG_FILE_PATH well! I don't know why the C lib function does't work in VS!  When put the search path directly, like vsg::Paths searchPaths = {"C:\VSG\vsgExamples\data"}; , Debugger goes normally, i.e., the examples can debug in visual studio

Thanks again !

**Reply by robertosfield** [reply]
Good to hear you are making progress.

I'm afraid I not use Windows + VisualStudio personally so can't advise on the specifics of setting environmental variables in the OS/VisualStudio.  Hopefully others with more knowledge on working with Windows can provide some pointers/

**Reply by ytf11** [reply]
@robertosfield

Understood! I can run and debug vsgExamples in visual studio with no problem now !!! it's enough for me. I hope I can use ur quality work in my project if I can !!! Progarmming Vulkan with its API is really hard and your work is very valuable like OSG even more !!! I may ask for your help in my feature work ...

Thank you very much !!!
